File: asset-management-test/src/main/java/com/assetmanagement/pages/LoginPage.java
package com.assetmanagement.pages;

import com.assetmanagement.utils.ConfigReader;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.time.Duration;

public class LoginPage {
    private static final Logger logger = LoggerFactory.getLogger(LoginPage.class);
    private final WebDriver driver;
    private final WebDriverWait wait;
    private final ConfigReader config;
    
    // 页面元素定位器
    private final By usernameInput = By.id("username");
    private final By passwordInput = By.id("password");
    private final By loginButton = By.id("loginBtn");
    private final By errorMessage = By.className("error-message");
    private final By dashboardElement = By.id("dashboard");
    
    public LoginPage(WebDriver driver) {
        this.driver = driver;
        this.config = new ConfigReader();
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(config.getTimeout()));
    }
    
    public void navigateToLoginPage() {
        driver.get(config.getBaseUrl());
    }
    
    public void login(String username, String password) {
        try {
            logger.info("尝试使用用户名 {} 登录", username);
            
            // 等待并填写用户名
            WebElement usernameField = wait.until(ExpectedConditions.presenceOfElementLocated(usernameInput));
            usernameField.clear();
            usernameField.sendKeys(username);
            
            // 填写密码
            WebElement passwordField = driver.findElement(passwordInput);
            passwordField.clear();
            passwordField.sendKeys(password);
            
            // 点击登录按钮
            driver.findElement(loginButton).click();
            logger.info("登录表单已提交");
            
        } catch (Exception e) {
            logger.error("登录过程中发生错误", e);
            throw e;
        }
    }
    
    public boolean isLoggedIn() {
        try {
            wait.until(ExpectedConditions.presenceOfElementLocated(dashboardElement));
            logger.info("成功检测到仪表板元素，用户已登录");
            return true;
        } catch (Exception e) {
            logger.info("未检测到仪表板元素，用户未登录");
            return false;
        }
    }
    
    public boolean isLoginErrorDisplayed() {
        try {
            WebElement error = wait.until(ExpectedConditions.presenceOfElementLocated(errorMessage));
            boolean isDisplayed = error.isDisplayed();
            logger.info("登录错误消息显示状态: {}", isDisplayed);
            return isDisplayed;
        } catch (Exception e) {
            logger.info("未检测到登录错误消息");
            return false;
        }
    }
} 

File: src/pages/login_page.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

class LoginPage:
    def __init__(self, url, username, password):
        self.url = url
        self.username = username
        self.password = password
        self.driver = None

    def start_browser(self):
        """启动浏览器并打开指定URL"""
        self.driver = webdriver.Edge()
        self.driver.get(self.url)
        self.driver.maximize_window()

    def login(self):
        """执行登录操作"""
        try:
            # 等待页面加载
            WebDriverWait(self.driver, 10).until(
                EC.presence_of_element_located((By.ID, "username"))
            )

            # 输入用户名
            username_input = self.driver.find_element(By.ID, "username")
            username_input.clear()
            username_input.send_keys(self.username)

            # 输入密码
            password_input = self.driver.find_element(By.ID, "password")
            password_input.clear()
            password_input.send_keys(self.password)

            # 检查是否有验证码
            try:
                captcha_input = self.driver.find_element(By.ID, "captcha")
                print("请手动输入验证码...")
                # 等待用户手动输入验证码
                while True:
                    captcha_value = captcha_input.get_attribute("value")
                    if captcha_value:
                        break
                    time.sleep(1)
            except:
                print("未检测到验证码输入框")

            # 点击登录按钮
            login_button = self.driver.find_element(By.ID, "login-button")
            login_button.click()

            # 等待登录成功
            WebDriverWait(self.driver, 10).until(
                EC.url_changes(self.url)
            )

            print("登录成功！")
            return True

        except Exception as e:
            print(f"登录失败: {str(e)}")
            return False

    def close(self):
        """关闭浏览器"""
        if self.driver:
            self.driver.quit()
